app/modules/auth/service.server.ts

```diff
@@ -299,6 +307,32 @@ export async function getAuthResponseByAccessToken(accessToken: string) {
   }
 }
 
+interface Session {
+  id: string;
+  revoked: boolean;
+}
+
+export async function validateSession(token: string) {
+  const t0 = performance.now();
+
+  const result = await db.$queryRaw<Partial<Session>[]>`
+      SELECT id, revoked FROM auth.refresh_tokens 
+      WHERE token = ${token} 
+      AND revoked = false
+      LIMIT 1 
+    `;
+  const t1 = performance.now();
+
+  // eslint-disable-next-line no-console
+  console.log(`Call to validateSession took ${t1 - t0} milliseconds.`);
+
+  if (result.length === 0) {
+    //logging for debug
+    Logger.error("Refresh token is invalid or has been revoked");
```

Comment:
Here you should log a shelf error, so we can see it in our errors logs.

Reply:
ack

Reply:
if (!isValidSession) {
        throw new ShelfError({
          cause: null,
          message: "Your session has expired. Please log in again.",
          label: "Auth",
        });
      }
      we are already doing it. this will log